src/app/portlets/dot-edit-page/content/components/dot-edit-page-state-controller/dot-edit-page-state-controller.component.ts

```diff
@@ -0,0 +1,259 @@
+import {
+    DotPersonalizeService,
+    DotCMSPersonalizedItem
+} from '@services/dot-personalize/dot-personalize.service';
+import { DotPageRenderOptions } from '@services/dot-page-render/dot-page-render.service';
+
+enum DotConfirmationType {
+    LOCK,
+    PERSONALIZATION
+}
+
+@Component({
+    selector: 'dot-edit-page-state-controller',
+    templateUrl: './dot-edit-page-state-controller.component.html',
+    styleUrls: ['./dot-edit-page-state-controller.component.scss']
+})
+export class DotEditPageStateControllerComponent implements OnInit, OnChanges {
+    @ViewChild('pageLockInfo') pageLockInfo: DotEditPageLockInfoComponent;
+
+    @Input() pageState: DotRenderedPageState;
+
+    lock: boolean;
+    mode: DotPageMode;
+    options$: Observable<SelectItem[]>;
+
+    private messages: { [key: string]: string } = {};
+
+    constructor(
+        private dotAlertConfirmService: DotAlertConfirmService,
+        private dotMessageService: DotMessageService,
+        private dotPageStateService: DotPageStateService,
+        private dotPersonalizeService: DotPersonalizeService
+    ) {}
+
+    ngOnInit() {
+        this.options$ = this.dotMessageService
+            .getMessages([
+                'editpage.toolbar.edit.page',
+                'editpage.toolbar.live.page',
+                'editpage.toolbar.preview.page',
+                'editpage.content.steal.lock.confirmation.message.header',
+                'editpage.content.steal.lock.confirmation.message'
+            ])
+            .pipe(
+                tap((messages: { [key: string]: string }) => {
+                    this.messages = messages;
+                }),
+                map(() => this.getStateModeOptions(this.pageState))
+            );
+    }
+
+    ngOnChanges(changes: SimpleChanges) {
+        this.setFieldsModels(changes.pageState.currentValue);
+    }
+
+    /**
+     * Handler locker change event
+     *
+     * @memberof DotEditPageToolbarComponent
+     */
+    lockPageHandler(): void {
+        if (this.shouldAskToLock()) {
+            this.showLockConfirmDialog().then(() => {
+                this.setLockerState();
+            });
+        } else {
+            this.setLockerState();
+        }
+    }
+
+    /**
+     * Handle the click to the locker switch
+     *
+     * @memberof DotEditPageStateControllerComponent
+     */
+    onLockerClick(): void {
+        if (!this.pageState.page.canLock) {
+            this.pageLockInfo.blinkLockMessage();
+        }
+    }
+
+    /**
+     * Handle state selector change event
+     *
+     * @param {DotPageMode} mode
+     * @memberof DotEditPageStateControllerComponent
+     */
+    stateSelectorHandler(mode: DotPageMode): void {
+        if (this.shouldShowConfirmation(mode)) {
+            this.lock = mode === DotPageMode.EDIT;
+
+            this.showConfirmation()
+                .then((type: DotConfirmationType) => {
+                    this.getPersonalize(type)
```

Comment:
```suggestion
if (DotConfirmationType.PERSONALIZATION) {
                    this.getPersonalize(type)
                    ...
```

even you can change all the subcribe code into the 'getPersonalize' method, of course the method name have to change